File: 0330-patching-array/0330-patching-array.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number}
 */
var minPatches = function(nums, n) {
	for (var sum = 0, idx = 0, added = 0; sum < n;) {
		added += idx === nums.length || nums[idx] > sum + 1 ? 1 : 0;
		sum += (nums[idx] || n) > sum + 1 ? sum + 1 : nums[idx++];
	}
	return added;  
};

File: 0330-patching-array/0330-patching-array.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        pat = 0
        obt = 0
        i = 0
        
        while obt < n:
            if i < len(nums) and nums[i] <= obt + 1:
                obt += nums[i]
                i += 1
            else:
                pat += 1
                obt += obt + 1
        return pat